libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -1035,4 +1081,37 @@ void AP_UAVCAN::update_mag_state(uint8_t node)
     }
 }
 
+bool AP_UAVCAN::led_write(uint8_t led_index, uint8_t red, uint8_t green, uint8_t blue) {
+    uavcan::equipment::indication::RGB565 color;
+    color.red = (red >> 3);
+    color.green = (green >> 2);
+    color.blue = (blue >> 3);
+    
+    if (!led_out_sem_take()) return false;
```

Comment:
This whole function looks or if order. The take should be first, then the array bounds check, then the shift. 
 Seems like the enable check is not even needed here

Reply:
Ok, I understand the enable check. That part is fine. Works be nice to have a comment saying the send happens above and it only computers in for registering new devices